common/testutils/test_utils.go

```diff
@@ -0,0 +1,74 @@
+package testutils
+
+import (
+	"fmt"
+	"github.com/stretchr/testify/assert"
+	"golang.org/x/exp/rand"
+	"testing"
+	"time"
+)
+
+// InitializeRandom initializes the random number generator. Prints the seed so that the test can be rerun
+// deterministically. Replace a call to this method with a call to initializeRandomWithSeed to rerun a test
+// with a specific seed.
+func InitializeRandom() {
```

Comment:
Could we always use `InitializeRandomWithSeed` and pass in random seed instead of having two separate methods?

Reply:
I've simplified this a little using varargs. You can now do `InitializeRandom()` or `InitializeRandom(1234)`.

This pattern is intentional. In general, you want to use a different random seed each time you test. But if you get a test failure, it's useful to go back and attempt a reproduction using the original seed that failed. In general, I wouldn't expect the caller to provide a seed unless they are attempting to debug a prior failed run.

Reply:
Discussed with Ian offline, he is ok with this as it currently is.